template_tests/allowmissingval.err

```diff
@@ -0,0 +1,3 @@
+{% macro foo() %} Foo is called {% endmacro %}
+This should {{ bar() }} error.
+This should throw error {{d.asd}}
```

Comment:
Name of variables can be more intuitive.

Reply:
Add tests for one more case.
1. {{d}} when d is not passed
2. {{ d.undefinedAttribute in struct d }} when d is passed but has no attribute `undefinedAttribute`
3. passing nil value explicitly for a var